### DISCUSSION TITLE
Is there a timeout for fileuploading?

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#20068] [Q&A]
When I upload my Berry Script (~27kb), it now fails almost every second time. Tasmota/Esp restarts and the edited file is empty. This has been happening since the file size exceeded approximately 15-17 kb. What could be the reason for this?
By the way: is there a better way to edit a Berry Script directly than to do copy and paste every time? VSC? FTP?

**Comment by sfromis** [comment]
If it is ESP restarting, it cannot be a timeout, but I'd not recommend copy/pasting that much. I read your words like you are doing this instead of actually uploading.

One way of handling files in a more graceful way would be the [UrlFetch ](https://tasmota.github.io/docs/Commands/#urlfetch) command, which is based on you having the file available on a web server.

Another viable approach is to push it using curl or similar:
`curl --form "ufsu=@/home/mypath/mycode.be" "http://tasmotadevice/ufsu"`

**Reply by ChrisVonFrosta** [reply]
i am on VSC on Windows :/ 
I have to look for a workaround

**Reply by ChrisVonFrosta** [reply]
ok, i see this works under windows promt, but not in powershell or VSC terminal

**Reply by ChrisVonFrosta** [reply]
but is much lesser pain then before to have a promt window open. thx

**Reply by sfromis** [reply]
You may have to tweak something to follow the rules of those, depending on how it fails.

Anyway, it is not required to specifically use `curl`, if you have other tools able to do a POST.
